modules/gateway/gateway.go

```diff
@@ -89,13 +89,15 @@ func (g *Gateway) Bootstrap(bootstrapPeer modules.NetAddress) (err error) {
 		return errors.New("couldn't learn hostname")
 	}
 
-	// request peers from the bootstrap
-	g.requestPeers(bootstrapPeer)
-	// save new peers
-	g.save()
+	// ask the bootstrapPeer to add us back
+	go g.RPC(bootstrapPeer, "AddMe", writerRPC(g.Address()))
 
-	// announce ourselves to the new peers
-	go g.threadedBroadcast("AddMe", writerRPC(g.myAddr))
+	// initial peer discovery
+	go func() {
```

Comment:
add some more comments here explaining what's happening and why it's happening that way instead of some other way.